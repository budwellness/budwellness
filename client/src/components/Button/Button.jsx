import React from 'react';
import PropTypes from 'prop-types';
import styles from './Button.module.scss';

function Button(props) {
  const {
    type, text, className, onClick,
  } = props;

  const buttonClasses = [
    styles.btn, styles[className],
    // className.includes('orangeBtn') ? styles.orangeBtn : '',
    // className.includes('whiteBtn') ? styles.whiteBtn : '',
  ].join(' ');

  return (
    // eslint-disable-next-line react/button-has-type
    <button type={type} className={buttonClasses} onClick={onClick}>
      {text}
    </button>
  );
}

Button.propTypes = {
  type: PropTypes.string,
  text: PropTypes.string,
  className: PropTypes.string,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  type: 'button',
  text: 'Our Products',
  className: 'btn',

  onClick: () => {},
};

export default Button;
