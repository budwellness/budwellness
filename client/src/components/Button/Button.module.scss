.btn {
  font-size: 17px;
  padding: 10px 25px;
  min-height: 48px;
  border-radius: 50px;
  font-weight: 700;
  font-family: $font-nunito;
  transition-timing-function: ease-in-out;
  transition-duration: 0.5s;
}

.orangeBtn {
  background-color: $main-orange;
  color: $main-green;
  border: none;
}

.orangeBtn:hover {
  transform: scale(1.1);
  cursor: pointer;
}

.whiteBtn {
  color: $secondary-green;
  border: 1px solid $secondary-green;
  background-color: $color-white;
}

.whiteBtn:hover {
  background-color: $secondary-green;
  color: $white-text-color;
  cursor: pointer;
}

.whiteBtn_active {
  color: $white-text-color;
  border: 1px solid $secondary-green;
  background-color: $secondary-green;
}
