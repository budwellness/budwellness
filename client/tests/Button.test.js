/* eslint-disable */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Button from '../src/components/Button/Button';

describe('Button testing snapshot', () => {
  test('should Button match snapshot', () => {
    const { asFragment } = render(
      <Button text="BUTTON" backgroundColor="aqua" type="button" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Button works', () => {
  test('should Button render with provided text', () => {
    render(<Button text="BUTTON" backgroundColor="aqua" type="button" />);
    expect(screen.getByText('BUTTON')).toBeInTheDocument();
  });
  test('should onClick function call when clicked', () => {
    const onClickMock = jest.fn();
    render(
      <Button
        text="BUTTON"
        backgroundColor="aqua"
        type="button"
        onClick={onClickMock}
      />
    );
    const buttonElem = screen.getByText('BUTTON');
    fireEvent.click(buttonElem);
    expect(onClickMock).toHaveBeenCalled();
  });
});
