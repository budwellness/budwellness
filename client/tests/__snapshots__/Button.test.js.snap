// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button testing snapshot should Button match snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn"
    type="button"
  >
    BUTTON
  </button>
</DocumentFragment>
`;
